sdk/scheduler/src/main/java/com/mesosphere/sdk/api/StateResource.java

```diff
@@ -94,6 +97,70 @@ public Response getPropertyKeys() {
         }
     }
 
+    @Path("/files")
+    @Produces(MediaType.TEXT_PLAIN)
+    @GET
+    public Response getFiles() {
+        try {
+            logger.info("Getting all files");
+            Collection<String> fileNames = StateStoreUtils.getFileNames(stateStore);
+            return ResponseUtils.plainOkResponse(fileNames.toString());
+        } catch (StateStoreException e) {
+            logger.error("Failed to get a list of files", e);
+            return Response.serverError().build();
+        }
+    }
+
+    @Path("/files/{file}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @GET
+    public Response getFile(@PathParam("file") String fileName) {
+        try {
+            logger.info("Getting file {}", fileName);
+            return ResponseUtils.plainOkResponse(StateStoreUtils.getFile(stateStore, fileName));
+        } catch (StateStoreException e) {
+            logger.error("Failed to get file {}: {}", fileName, e);
+            return ResponseUtils.plainResponse(
+                    String.format("Failed to get the file"),
+                    Response.Status.NOT_FOUND
+            );
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Cannot encode data: ", e);
+            return Response.serverError().build();
+        }
+    }
+
+    @Path("/files/{file}")
+    @PUT
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response putFile(
+            @FormDataParam("file") InputStream uploadedInputStream,
+            @FormDataParam("file") FormDataContentDisposition fileDetails
+    ) {
+        logger.info(fileDetails.toString());
+        String fileName = fileDetails.getFileName();
+        if (uploadedInputStream == null || fileName == null) {
+            return ResponseUtils.plainResponse(
+                    "Specify 'file' as a parameter in your request",
+                    Response.Status.BAD_REQUEST
+            );
+        }
+
+        try {
+            if (stateStore.hasProperty(StateStoreUtils.FILE_NAME_PREFIX + fileName)) {
+                String msg = String.format("File %s is already in the state store", fileName);
+                logger.info(msg);
+                return ResponseUtils.plainOkResponse(msg);
+            }
+            logger.info("Storing {}", fileName);
```

Comment:
Should have a sanity check against the metadata that the advertised size isn't too big (maybe a field in fileDetails has this?)